### PR TITLE
pre-alpha Release [S-111249, S-111251] -  Release integration with List of Release Templates and execution from Backstage plugin

### PR DESCRIPTION
**Pull request by sishwarya**
## Release Checklist

- All steps below must be completed before merging.

## Changes Made

The below feature has been added to the backstage release plugin.
- Monitor release templates.
- Link to create a new release.
- Link to view the Templates.
- Permission feature to view a plugin
- Multiple Instance support

### Required

- [x] PR title should follow correct format.
  - MAJOR Release is used when there are breaking changes involved
  - MINOR Release is used when enhancements are introduced
  - PATCH Release is used for defects, security, tech chores, etc.
  - pre-alpha is used for testing releases
  - Note:
    - Based on the commit messages, increment the version from the latest release.
    - Push the bumped npm version in package.json back into the repo.
    - Push a tag for the new version back into the repo.
    - [For more details on versioning](https://github.com/phips28/gh-action-bump-version/blob/master/README.md) 
- [x] Checklist of **changes made** added to PR description
- [x] Related issue(s) linked to PR

 
## Reviewer Checklist

- [ ] Author checklist has been reviewed?
- [ ] All acceptance criteria has been met from linked issue
- [ ] Run code locally and verify all changes

## Related Issue(s)

Resolves #XXXX
Story-
S-111249 - Frontend Plugin Changes - Templates screen
S-111251 - Backend Plugin Changes - Templates
S-111254 - Frontend Plugin Changes - Multi-Instance Support
Defect-######

### CODE REVIEW

File: plugins/dai-release/src/components/TemplateHomePageComponent/TemplateHomePageComponent.tsx [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+import { ReleasePopOverComponent } from '../ReleasePopOverComponent';
+import { ReleaseResponseErrorPanel } from '../ReleaseResponseErrorPanel';
+import { SearchHeaderComponent } from '../SearchHeaderComponent';
+import Typography from '@mui/material/Typography';
+import capitalize from 'lodash/capitalize';
+import { createSvgIcon } from '@mui/material/utils';
+import releaseLogoWhite from '../../assets/releaseLogoWhite.png';
+import { useTemplates } from '../../hooks';
+
+const PlusIcon = createSvgIcon(
```
Comment (Karthigaiselvan) [inline plugins/dai-release/src/components/TemplateHomePageComponent/TemplateHomePageComponent.tsx:17]: Can we move this plusIcon in any common place. which is used in densetable too.

File: plugins/dai-release-backend/src/api/apiConfig.ts [hunk 2]
```diff
@@ -47,6 +51,26 @@ export const getReleaseDetailsRedirectUri = (
   return `${getReleaseApiHost(config)}${RELEASE_DETAILS_REDIRECT_PATH}${releaseIdUrl}`;
 };
 
+export const getTemplateDetailsRedirectUri = (
```
[the remark this answers is not in the document]
Comment (sishwarya) [reply plugins/dai-release-backend/src/api/apiConfig.ts:54]: @Karthigaiselvan , Both have different URI to redirect, so I created two different method to handle it

File: plugins/dai-release-backend/src/api/ReleaseOverviewApi.ts [hunk 3]
```diff
@@ -257,4 +262,68 @@ export class ReleaseOverviewApi {
   async getReleaseInstances() {
     return this.config.getInstanceList();
   }
+
```
Comment (nelson-digi) [inline plugins/dai-release-backend/src/api/ReleaseOverviewApi.ts:265]: Can be added in separate file. TemplatesApi.ts and move FolderApis to FolderApi.ts file and its test cases. Created a story for that along with refresh button.

https://www7.v1host.com/V1Production/assetdetail.v1?number=S-113627

File: plugins/dai-release/src/components/TemplateHomePageComponent/TemplateHomePageComponent.tsx (hunk 1 above)
Comment (sishwarya) [reply plugins/dai-release/src/components/TemplateHomePageComponent/TemplateHomePageComponent.tsx]: done

**Pull request merged**